com.unity.netcode.gameobjects/Tests/Runtime/NetworkVariableUserSerializableTypesTests.cs

```diff
@@ -95,19 +89,20 @@ public IEnumerator WhenUsingAUserSerializableNetworkVariableWithUserSerializatio
                 value = new MyTypeOne();
                 reader.ReadValueSafe(out value.Value);
             };
-            var serverObject = Object.Instantiate(m_WorkingPrefab);
+
+            var serverObject = SpawnObject(m_WorkingPrefab, m_ServerNetworkManager);
             var serverNetworkObject = serverObject.GetComponent<NetworkObject>();
-            serverNetworkObject.NetworkManagerOwner = m_ServerNetworkManager;
-            serverNetworkObject.Spawn();
 
-            yield return NetcodeIntegrationTestHelpers.WaitForMessageOfTypeReceived<CreateObjectMessage>(m_ClientNetworkManagers[0]);
+            yield return WaitForMessageOfType<CreateObjectMessage>(m_ClientNetworkManagers[0]);
 
-            serverNetworkObject.GetComponent<WorkingUserNetworkVariableComponent>().NetworkVariable.Value = new MyTypeOne { Value = 20 };
+            var serverComponent = serverObject.GetComponent<WorkingUserNetworkVariableComponent>();
+            serverComponent.NetworkVariable.Value = new MyTypeOne { Value = 20 };
+            var clientObject = GetComponentForClient<WorkingUserNetworkVariableComponent>(m_ClientNetworkManagers[0].LocalClientId);
 
-            yield return NetcodeIntegrationTestHelpers.WaitForMessageOfTypeReceived<NetworkVariableDeltaMessage>(m_ClientNetworkManagers[0]);
-            yield return NetcodeIntegrationTestHelpers.WaitForMessageOfTypeReceived<NetworkVariableDeltaMessage>(m_ClientNetworkManagers[0]);
+            yield return WaitForMessageOfType<NetworkVariableDeltaMessage>(m_ClientNetworkManagers[0], 2);
 
-            var clientObject = GetComponentForClient<WorkingUserNetworkVariableComponent>(m_ClientNetworkManagers[0].LocalClientId);
+            yield return WaitForConditionOrTimeOut(() => serverComponent.NetworkVariable.Value.Value == clientObject.NetworkVariable.Value.Value);
```

Comment:
This wait shouldn't be needed. After waiting for two NetworkVariableDeltaMessages, this condition should be true. If it's not, the test has failed and there's no need to wait further.